flink-runtime/src/main/java/org/apache/flink/runtime/state/SharedStateRegistry.java

```diff
@@ -106,4 +108,12 @@ StreamStateHandle registerReference(
     void registerAllAfterRestored(CompletedCheckpoint checkpoint, RestoreMode mode);
 
     void checkpointCompleted(long checkpointId);
+
+    /** Create a unique key based on physical id. */
+    static SharedStateRegistryKey createSharedStateRegistryKey(StreamStateHandle handle) {
+        String keyString = handle.getStreamStateHandleID().getKeyString();
+        // key strings tend to be longer, so we use the MD5 of the key string to save memory
+        return new SharedStateRegistryKey(
+                UUID.nameUUIDFromBytes(keyString.getBytes(StandardCharsets.UTF_8)).toString());
+    }
```

Comment:
Static factory methods are usually placed in the type which instance they create,
so how about moving this method to `SharedStateRegistryKey`?

Reply:
Thanks for the guidance, fixed it.